spec/calendars/calendarevent.mdown

```diff
@@ -21,18 +21,18 @@ Standard "/changes" method
 
 ## CalendarEvent/set
 
-Standard "/set" method.
-
-When an event is created, updated or destroyed, the server MUST also ensure the following:
-
-- Any alerts are scheduled/cancelled correctly.
-- If there is a *participantId*, and the corresponding participant has a *role*
-  of `owner`:
-  - If an event is created/updated: send a REQUEST iMIP email with the event as
-    an ICS attachment to all participants that are not "you".
-  - When an event is destroyed, if it is in the future, then email all
-    participants other than you the appropriate iMIP email to inform them that the event has been cancelled. If it is in the past, the server SHOULD NOT send a message.
-- If there is a *participantId*, and the corresponding participant does not have a *role* of `owner`, and the *scheduleStatus* is updated for this participant, send the appropriate iMIP email to the *replyTo* address.
+Standard "/set" method, with the following extra argument:
+
+- **onSuccessScheduleCalendarEvent**:  Id[]|null
+  A list of *CalendarEvent ids* for which iTIP messages should be sent, if the create/update/destroy of the CalendarEvent object with the corresponding id succeeds.
```

Comment:
What if the change is irrelevant (e.g. you change your alerts for an event you have already RSVPed to)? If you include the id in onSuccessScheduleCalendarEvent, will it send out another iTIP message identical to the previous one? (I guess that's OK, the client just needs to do more calculations about whether to send it, but I think it should be explicitly stated that this will happen.)

Reply:
Another thing: suppose I receive an iMIP response email. Now I update my calendar with the data from that response using JMAP (so I update the participant's participation status, sequence number etc.). I may want to broadcast an iMIP message with this change to all the other participants but only if they use iSchedule rather than iMIP. There's no way to specify this (either that the change shouldn't be sent back to the person who just sent my their RSVP, or that it should only be sent to those with a back channel that doesn't go via email.)

If it's implicit on the server then the server can be smart about it.